pkg/render/logstorage.go

```diff
@@ -1705,6 +1751,168 @@ func (es elasticsearchComponent) oidcUserRoleBinding() []client.Object {
 	}
 }
 
+func (es elasticsearchComponent) esGatewayDeployment() *appsv1.Deployment {
+	envVars := []corev1.EnvVar{
+		{Name: "ES_GATEWAY_LOG_LEVEL", Value: "DEBUG"},
+		{Name: "ES_GATEWAY_ELASTIC_ENDPOINT", Value: ElasticsearchHTTPSEndpoint},
+		{Name: "ES_GATEWAY_KIBANA_ENDPOINT", Value: KibanaHTTPSEndpoint},
+		{Name: "ES_GATEWAY_HTTPS_CERT", Value: "/certs/https/tls.crt"},
+		{Name: "ES_GATEWAY_HTTPS_KEY", Value: "/certs/https/key"},
+	}
+
+	dnsSvcNames := dns.GetServiceDNSNames(EsGatewayElasticServiceName, ElasticsearchNamespace, es.clusterDomain)
+	dnsSvcNames = append(dnsSvcNames, dns.GetServiceDNSNames(EsGatewayKibanaServiceName, KibanaNamespace, es.clusterDomain)...)
+	var initContainers []corev1.Container
+	if es.installation.CertificateManagement != nil {
+		initContainers = append(initContainers, CreateCSRInitContainer(
+			es.installation.CertificateManagement,
+			es.csrImage,
+			EsGatewayElasticPublicCertSecret,
+			EsGatewayElasticServiceName,
+			EsGatewaySecretKeyName,
+			EsGatewaySecretCertName,
+			dnsSvcNames,
+			ElasticsearchNamespace))
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: EsGatewayVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: EsGatewayElasticPublicCertSecret,
+				},
+			},
+		},
+		{
+			Name: KibanaPublicCertSecret,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: KibanaPublicCertSecret,
+				},
+			},
+		},
+		{
+			Name: relasticsearch.PublicCertSecret,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: relasticsearch.PublicCertSecret,
+				},
+			},
+		},
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{Name: EsGatewayVolumeName, MountPath: "/certs/https", ReadOnly: true},
+		{Name: KibanaPublicCertSecret, MountPath: "/certs/kibana", ReadOnly: true},
+		{Name: relasticsearch.PublicCertSecret, MountPath: "/certs/elasticsearch", ReadOnly: true},
+	}
+
+	podTemplate := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      EsGatewayName,
+			Namespace: ElasticsearchNamespace,
+			Labels: map[string]string{
+				"k8s-app": EsGatewayName,
+			},
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: EsGatewayServiceAccountName,
+			ImagePullSecrets:   secret.GetReferenceList(es.pullSecrets),
+			InitContainers:     initContainers,
+			Volumes:            volumes,
+			Containers: []corev1.Container{
+				{
+					Name:          EsGatewayName,
+					Image:         es.esGatewayImage,
+					Env:           envVars,
+					LivenessProbe: complianceLivenessProbe,
```

Comment:
This looks like a typo. 

We will have a readiness probe but we probably won't define a liveness probe.